python/google/protobuf/pyext/descriptor.cc

```diff
@@ -123,6 +119,10 @@ bool _CalledFromGeneratedFile(int stacklevel) {
     PyErr_Clear();
     return false;
   }
+  if ((filename_size < 3) || (strcmp(&filename[filename_size - 3], ".py") != 0)) {
```

Comment:
Could we use C++ string manipulation instead? `strcmp()` scares me.

Reply:
The original code is using strcmp. I also searched the code base, strcmp has been used several times. If we want to use string manipulation instead, I'd prefer to in another change.